### PR TITLE
fix(cardinality): Records with overflow doesn't pass dataset + Aggregation issue in TsCardReduceExec

### PR DESCRIPTION
**Pull request by sandeep6189**
**Current behavior :**

PR fixes two issues:

1. Cardinality records with `_overflow_` prefix doesn't pass dataset as expected by the QueryService in V2 APIs. 

2. Correctness issue: Incorrect aggregation when the hashmap reaches the map size. As of today, we would skip the keys already present in the map and always aggregate the overflow bucket.

**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**New behavior :**
The fixes and unit tests are added for the above issues.

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala [hunk 1]
```diff
@@ -106,8 +119,17 @@ final case class TsCardReduceExec(queryContext: QueryContext,
...
                                  firstSchema: Task[ResultSchema],
                                  querySession: QuerySession): Observable[RangeVector] = {
-    val taskOfResults = childResponses.flatMap(res => Observable.fromIterable(res._1.result))
-      .foldLeftL(new mutable.HashMap[ZeroCopyUTF8String, CardCounts])(mapFold)
+
+    // capture the result schema of responses
+    var rs = ResultSchema.empty
+    val flatMapData = childResponses.flatMap(res => {
+      if (rs == ResultSchema.empty) {
+        rs = res._1.resultSchema
```
Comment (yu-shipit) [inline query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala:127]: do all child responses have the same schema?

**Review by kvpetrov — APPROVED** [review]
LGTM

File: query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala (hunk 1 above)
Comment (sandeep6189) [reply query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala:127]: Yep, for the same version, each of them will have same schema

File: query/src/test/scala/filodb/query/exec/MetadataExecSpec.scala [hunk 2]
```diff
@@ -434,4 +435,64 @@ class MetadataExecSpec extends AnyFunSpec with Matchers with ScalaFutures with B
       }
     }
   }
+
+  it ("should add overflow group") {
```
Comment (alextheimer) [inline query/src/test/scala/filodb/query/exec/MetadataExecSpec.scala:439]: A nontrivial test for a relatively simple fix -- the effort here is much appreciated

**Review by alextheimer — APPROVED** [review]
Thanks for this ultra-quick fix @sandeep6189

**Pull request merged**